### DISCUSSION TITLE
Use Cases for Minting tokens in Marlowe contracts

### DISCUSSION THREAD

**Discussion by bwbush** [input-output-hk/MIPs#12] [Marlowe Improvements]
Describe the specifics of potential use cases for minting native tokens in Marlowe contracts.

**Comment by apestchanker** [comment]
We are using Marlowe for a VC FUND collecting contract, which is sort of a Crowdfunding style contract.
Each investor will send Tokens to the Contract in a self selected amount (via CHOICE action), when the contract reaches its deadline, two things can happen, the contract surpassed the Threshold or reached its Target amount, then it is declared Fully Funded, or the deposited tokens will be sent back to investors on the contrary.
If the contract is fully funded, we need to mint an ID NFT that will represent the investors rights to claim the Fund's returns (after investing them in companies) and that ID NFT will contain in its metadata the amount invested by the investor, the date and other data to make it verifiable, so when claiming, the proper percentage is known and allows the holder to claim its well deserved returns.

Right now we are going to mint those NFTs outside the contract (After closing), but it would be awesome to be able to make the contract mint them, if the contract gets fully funded.

Hope that is a good case that can be someday found within Marlowe.

Thanks for reading!
